File: features/merchant/review/src/main/java/com/tokopedia/review/feature/inbox/buyerreview/view/activity/InboxReputationFilterActivity.java
package com.tokopedia.review.feature.inbox.buyerreview.view.activity;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity;
import com.tokopedia.design.text.TextDrawable;
import com.tokopedia.review.R;
import com.tokopedia.review.feature.inbox.buyerreview.view.fragment.InboxReputationFilterFragment;
import com.tokopedia.review.feature.inbox.buyerreview.view.fragment.InboxReputationFragment;

/**
 * @author by nisie on 8/21/17.
 */

public class InboxReputationFilterActivity extends BaseSimpleActivity {

    public interface ResetListener {
        void resetFilter();
    }

    ResetListener listener;

    public static Intent createIntent(Context context, String timeFilter,
                                      String scoreFilter,
                                      int tab) {
        Intent intent = new Intent(context, InboxReputationFilterActivity.class);
        intent.putExtra(InboxReputationFilterFragment.SELECTED_TIME_FILTER, timeFilter);
        intent.putExtra(InboxReputationFilterFragment.SELECTED_SCORE_FILTER, scoreFilter);
        intent.putExtra(InboxReputationFragment.PARAM_TAB, tab);
        return intent;
    }

    @Nullable
    @Override
    protected Fragment getNewFragment() {
        String timeFilter = getIntent().getStringExtra(InboxReputationFilterFragment
                .SELECTED_TIME_FILTER);
        String statusFilter = getIntent().getStringExtra(InboxReputationFilterFragment
                .SELECTED_SCORE_FILTER);
        int tab = getIntent().getIntExtra(InboxReputationFragment
                .PARAM_TAB, -1);

        Fragment fragment = InboxReputationFilterFragment.createInstance(timeFilter, statusFilter, tab);
        listener = (InboxReputationFilterFragment) fragment;

        return fragment;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        menu.add(Menu.NONE, R.id.action_reset, 0, "");
        MenuItem menuItem = menu.findItem(R.id.action_reset);
        menuItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS);
        menuItem.setIcon(getResetMenu());
        return true;
    }

    private Drawable getResetMenu() {
        TextDrawable drawable = new TextDrawable(this);
        drawable.setText(getResources().getString(R.string.reset_title));
        drawable.setTextColor(com.tokopedia.unifyprinciples.R.color.Unify_N700_68);
        return drawable;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_reset
                && listener != null) {
            listener.resetFilter();
            return true;
        }
        else {
            return super.onOptionsItemSelected(item);
        }
    }



    @Override
    protected void setupLayout(Bundle savedInstanceState) {
        super.setupLayout(savedInstanceState);
        toolbar.setPadding(0,0,20,0);
    }
}

File: features/merchant/review_inbox/src/main/java/com/tokopedia/review/feature/inbox/buyerreview/view/activity/InboxReputationFilterActivity.kt
package com.tokopedia.review.feature.inbox.buyerreview.view.activity

import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.fragment.app.Fragment
import com.tokopedia.abstraction.base.view.activity.BaseSimpleActivity
import com.tokopedia.design.text.TextDrawable
import com.tokopedia.review.feature.inbox.buyerreview.view.fragment.InboxReputationFilterFragment
import com.tokopedia.review.feature.inbox.buyerreview.view.fragment.InboxReputationFragment
import com.tokopedia.review.inbox.R

/**
 * @author by nisie on 8/21/17.
 */
class InboxReputationFilterActivity : BaseSimpleActivity() {

    interface ResetListener {
        fun resetFilter()
    }

    var listener: ResetListener? = null

    override fun getNewFragment(): Fragment {
        val timeFilter: String? =
            intent.getStringExtra(InboxReputationFilterFragment.SELECTED_TIME_FILTER)
        val statusFilter: String? =
            intent.getStringExtra(InboxReputationFilterFragment.SELECTED_SCORE_FILTER)
        val tab: Int = intent.getIntExtra(InboxReputationFragment.PARAM_TAB, -1)
        val fragment: Fragment =
            InboxReputationFilterFragment.createInstance(timeFilter, statusFilter, tab)
        listener = fragment as InboxReputationFilterFragment?
        return fragment
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menu.add(Menu.NONE, R.id.action_reset, 0, "")
        val menuItem: MenuItem = menu.findItem(R.id.action_reset)
        menuItem.setShowAsAction(MenuItem.SHOW_AS_ACTION_ALWAYS)
        menuItem.icon = resetMenu
        return true
    }

    private val resetMenu: Drawable
        get() {
            val drawable = TextDrawable(this)
            drawable.text = resources.getString(R.string.reset_title)
            drawable.setTextColor(com.tokopedia.unifyprinciples.R.color.Unify_N700_68)
            return drawable
        }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return if ((item.itemId == R.id.action_reset && listener != null)) {
            listener?.resetFilter()
            true
        } else {
            super.onOptionsItemSelected(item)
        }
    }

    override fun setupLayout(savedInstanceState: Bundle?) {
        super.setupLayout(savedInstanceState)
        toolbar.setPadding(PADDING_ZERO, PADDING_ZERO, PADDING_20, PADDING_ZERO)
    }

    companion object {
        const val PADDING_20 = 20
        const val PADDING_ZERO = 0

        fun createIntent(
            context: Context?, timeFilter: String?,
            scoreFilter: String?,
            tab: Int
        ): Intent {
            val intent = Intent(context, InboxReputationFilterActivity::class.java)
            intent.putExtra(
                InboxReputationFilterFragment.SELECTED_TIME_FILTER,
                timeFilter
            )
            intent.putExtra(
                InboxReputationFilterFragment.SELECTED_SCORE_FILTER,
                scoreFilter
            )
            intent.putExtra(InboxReputationFragment.PARAM_TAB, tab)
            return intent
        }
    }
}